plugins/woocommerce/includes/wc-product-functions.php

```diff
@@ -392,7 +392,7 @@ function wc_get_formatted_variation( $variation, $flat = false, $include_names =
 			// If this is a term slug, get the term's nice name.
 			if ( taxonomy_exists( $name ) ) {
 				$term = get_term_by( 'slug', $value, $name );
-				if ( ! is_wp_error( $term ) && ! empty( $term->name ) ) {
+				if ( ! is_wp_error( $term ) && '' !== $term->name ) {
```

Comment:
Sorry I haven't noticed this before, but the unit tests showed that `empty` actually behaves in a way that `$term` can be `null` and it won't produce a PHP Notice while this new code does, so we'd first need to test whether `$term` is set. Can you please update the code to include the check? Thanks!